src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,159 @@
+package duke.parser;
+
+import java.time.LocalDate;
+
+import duke.exception.DeadlineException;
+import duke.exception.DukeException;
+import duke.exception.EventException;
+import duke.exception.ToDoException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+
+public class Parser {
+    private final TaskList taskList;
+    private final Ui ui;
+
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+        this.ui = new Ui();
+    }
+
+    /**
+     * Returns true if the program should still keep going on (e.g., when input is not equal to "bye")
+     *
+     * @param input full raw user input line.
+     * @return true if input is not equal to "bye".
+     */
+    public boolean parseInput(String input) {
+        ui.printDivider();
+        try {
+            if (input.equals("bye")) {
+                ui.print("Goodbye human. See you soon!");
+                return false;
+            } else if (input.equals("list")) {
+                printItems();
+            } else if (input.contains("done")) {
+                markAsDone(input);
+            } else if (input.contains("delete")) {
+                deleteItem(input);
+            } else if (input.contains("find")) {
+                findItem(input);
+            } else {
+                addItem(input);
+            }
+        } catch (DukeException e) {
+            ui.print(e.getMessage());
+        }
+
+        ui.printDivider();
+        return true;
+    }
+
+    /**
+     * Parses add command from the user.
+     *
+     * @param input full raw user input line.
+     * @throws DukeException if the user input is of an invalid format.
+     */
+    private void addItem(String input) throws DukeException {
+        Task newItem = null;
+        if (input.contains("todo")) {
```

Comment:
Unless I am misunderstanding your code, if you leave it as `input.contains("todo")`, wouldn't there be a bug where the user types, for example, "bla bla todo homework"? The if-condition is fulfilled, but the outcome might not be as expected. Same for other else if statements below.

Reply:
Yeah, theres a .startsWith(<string>) method for String classes in Java!